src/string_search.h

```diff
@@ -0,0 +1,597 @@
+// Copyright 2011 the V8 project authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SRC_STRING_SEARCH_H_
+#define SRC_STRING_SEARCH_H_
+
+#include "node.h"
+#include <string.h>
+
+namespace node {
+namespace stringsearch {
+
+// Returns the maximum of the two parameters.
+template <typename T> T Max(T a, T b) { return a < b ? b : a; }
+
+static const uint32_t kMaxOneByteCharCodeU = 0xff;
+
+static inline int NonOneByteStart(const uint16_t *chars, int length) {
+  const uint16_t *limit = chars + length;
+  const uint16_t *start = chars;
```

Comment:
It would make sense to add a custom `.clang-format` to make it easier to comply with the coding standards.

http://www.chromium.org/developers/coding-style
I used `.clang-format` from Chromium and it is the same style you prefer because the normal Google style does not enforce * and & by the type rather than the variable name. 
````
# Defines the Chromium style for automatic reformatting.
# http://clang.llvm.org/docs/ClangFormatStyleOptions.html
BasedOnStyle: Chromium
# This defaults to 'Auto'. Explicitly set it for a while, so that
# 'vector<vector<int> >' in existing files gets formatted to
# 'vector<vector<int>>'. ('Auto' means that clang-format will only use
# 'int>>' if the file already contains at least one such instance.)
Standard: Cpp11
````